SU2_CFD/include/solvers/CTurbSolver.hpp

```diff
@@ -147,4 +147,113 @@ class CTurbSolver : public CScalarSolver<CTurbVariable> {
    * \returns The number of extra variables.
    */
   unsigned long RegisterSolutionExtra(bool input, const CConfig* config) final;
+  
+  /*!
+   * \brief Compute the viscous flux for the turbulence equations at a particular edge in a non-conservative manner.
+   * \tparam SolverSpecificNumericsTemp - lambda-function, to implement solver specific contributions to numerics.
+   * \note The functor has to implement (iPoint, jPoint)
+   * \param[in] iEdge - Edge for which we want to compute the flux
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] solver_container - Container vector with all the solutions.
+   * \param[in] numerics - Description of the numerical method.
+   * \param[in] config - Definition of the particular problem.
+   */
+  inline virtual void Viscous_Residual(const unsigned long iEdge, const CGeometry* geometry, CSolver** solver_container,
+                                     CNumerics* numerics, const CConfig* config) {}
+  template<typename SolverSpecificNumericsTemp>
+  void TurbViscousResidual(const unsigned long iEdge, const CGeometry* geometry, CSolver** solver_container, 
+                                        CNumerics* numerics, const CConfig* config, SolverSpecificNumericsTemp&& SolverSpecificNumerics);
+
+  /*!
+   * \brief Compute a suitable under-relaxation parameter to limit the change in the solution variables over
+   * a nonlinear iteration for stability.
+   * \param[in] allowableRatio - Maximum percentage update in variable per iteration.
+   */
+  void ComputeUnderRelaxationFactorHelper(su2double allowableRatio);
 };
+
+template<typename SolverSpecificNumericsTemp>
+void CTurbSolver::TurbViscousResidual(const unsigned long iEdge, const CGeometry* geometry, CSolver** solver_container,
```

Comment:
Move to where we have the other generic implementation and rename to reflect the non-conservative nature